src/products/components/ProductUpdatePage/ProductChannelsListingsDialog.tsx

```diff
@@ -0,0 +1,72 @@
+import ChannelsAvailabilityDialog from "@saleor/components/ChannelsAvailabilityDialog";
+import { ChannelFragment } from "@saleor/graphql";
+import useStateFromProps from "@saleor/hooks/useStateFromProps";
+import { DialogProps } from "@saleor/types";
+import { arrayDiff } from "@saleor/utils/arrays";
+import { toggle } from "@saleor/utils/lists";
+import React from "react";
+import { useIntl } from "react-intl";
+
+import { ProductUpdateData } from "./form";
+
+export type ProductChannelsListingDialogSubmit = (
+  update: Record<"added" | "removed", string[]>,
+) => void;
+
+export interface ProductChannelsListingsDialogProps extends DialogProps {
+  channels: ChannelFragment[];
+  data: Pick<ProductUpdateData, "channels">;
+  onConfirm: ProductChannelsListingDialogSubmit;
+}
+
+const ProductChannelsListingsDialog: React.FC<ProductChannelsListingsDialogProps> = ({
+  channels,
+  data,
+  open,
+  onClose,
+  onConfirm,
+}) => {
+  const intl = useIntl();
+
+  const [selected, setSelected] = useStateFromProps(
+    data.channels.updateChannels.map(listing => listing.channelId),
+  );
```

Comment:
I think we shouldn't use useStateFromProps in new components, it should be depreciated in favor of useEffect which does reference comparisons together with memoized initial value

Reply:
I know, but didn't want to go into rabbit hole here (as my time here is limited), so I just reused old code